base/inference.jl

```diff
@@ -63,6 +63,13 @@ immutable Const
     Const(v::ANY) = new(v)
 end
 
+immutable PartialTypeVar
+    tv::TypeVar
+    lb_certain::Bool
+    ub_certain::Bool
```

Comment:
If I'm not mistaken, Keno is using the identity of the TypeVar to infer the equivalence class of the result of `UnionAll(tv, apply_type(Type, tv)))`